packages/angular/projects/clr-angular/src/data/datagrid/datagrid-row.spec.ts

```diff
@@ -121,6 +121,40 @@ export default function (): void {
       });
     });
 
+    describe('Conditional selection switch order or arguments', function () {
+      let context: TestContext<ClrDatagridRow, SelectableRowOrder>;
+      let selectionProvider: Selection;
+      let checkbox: HTMLElement;
+
+      beforeEach(function () {
+        context = this.create(ClrDatagridRow, SelectableRowOrder, DATAGRID_SPEC_PROVIDERS);
+        selectionProvider = TestBed.get(Selection);
+        TestBed.get(Items).all = [{ id: 1 }, { id: 2 }];
+      });
+
+      it('should toggle when clrDgSelectable is false for type  SelectionType.Multi', () => {
+        selectionProvider.selectionType = SelectionType.Multi;
+        context.testComponent.clrDgSelectable = false;
+        context.detectChanges();
+        checkbox = context.clarityElement.querySelector("input[type='checkbox']");
+
+        expect(checkbox.getAttribute('disabled')).toBe('true');
+        expect(checkbox.getAttribute('aria-disabled')).toBe('true');
+
+        context.clarityDirective.toggle();
+        expect(context.clarityDirective.selected).toBe(true);
+      });
+
+      it('should be able you toggle the state of selection when clrDgSelectable is false', () => {
```

Comment:
nit: should be able to toggle selection through API when clrDgSelectable is false